astropy/io/fits/tests/test_core.py

```diff
@@ -1015,6 +1017,63 @@ def test_filename_with_colon(self):
         with fits.open(self.temp(filename)) as hdul:
             assert np.all(hdul[0].data == hdu.data)
 
+
+    def test_writeto_full_disk(self, monkeypatch):
+        """
+        Test that it gives a readable error when trying to write an hdulist
+        to a full disk.
+        """
+        def _writeto(self, array):
+            raise OSError("Fake error raised when writing file.")
+
+        def get_free_space_in_dir(path):
+            return 0
```

Comment:
What if the free space is `>0` but just not enough? That does not seem to be tested.